abci/example/kvstore/kvstore.go

```diff
@@ -166,34 +167,64 @@ func (app *Application) InitChain(_ context.Context, req *types.InitChainRequest
 // - `=` is not the first or last byte.
 // - if key is `val` that the validator update transaction is also valid.
 func (app *Application) CheckTx(_ context.Context, req *types.CheckTxRequest) (*types.CheckTxResponse, error) {
+	code := CodeTypeOK
 	// If it is a validator update transaction, check that it is correctly formatted
 	if isValidatorTx(req.Tx) {
 		if _, _, _, err := parseValidatorTx(req.Tx); err != nil {
-			if app.useLanes {
-				return &types.CheckTxResponse{Code: CodeTypeInvalidTxFormat, Lane: app.lanes["val"]}, nil
```

Comment:
You could leave this for validator txs to always have a high priority, these txs should anyways not be of the form '2=2' , right?

Reply:
These txs have the form `val=...`.

Reply:
Exactly so for those we could already return here, no need to do the other stuff.

Reply:
Below there's a check for `!app.useLanes`. If it holds, we don't have to return a lane.